src/main/scala/com/dataartisans/flinktraining/exercises/dataset_scala/tf_idf/MailTFIDF.scala

```diff
@@ -37,137 +32,57 @@ import scala.collection.mutable
-        if (!stopWords.contains(word) && wordPattern.matcher(word).matches) {
-          // word candidate is not a stop word and matches the word pattern
-          // increase word count
-          val cnt = wordCounts.getOrElse(word, 0)
-          wordCounts.put(word, cnt+1)
-        }
-      }
-      // emit all word counts per document and word
-      for (wc <- wordCounts.iterator) {
-        out.collect( (t._1, wc._1, wc._2) )
-      }
-    }
-  }
+  // word pattern regular expression
+  private val WORD_PATTERN = "(\\p{Alpha})+".r
 
-  /**
-   * Extracts the unique words in a mail.
-   * Words consist only of alphabetical characters. Frequent words (stop words) are filtered out.
-   *
-   * @param stopWordsA  Array of words that are filtered out.
-   */
-  class UniqueWordExtractor(stopWordsA: Array[String])
-    extends FlatMapFunction[(String, String), (String, Int) ] {
-
-    val stopWords = new mutable.HashSet[String]
-    val uniqueWords = new mutable.HashSet[String]
-    // initalize pattern to match words
-    val wordPattern: Pattern = Pattern.compile("(\\p{Alpha})+")
-
-    // initialize set of stop words
-    for(sw <- stopWordsA) {
-      this.stopWords.add(sw)
-    }
-
-    override def flatMap(t: (String, String), out: Collector[(String, Int)]): Unit = {
-      // clear unique words
-      uniqueWords.clear()
-
-      // split mail along whitespaces
-      val tokens = new StringTokenizer(t._2)
-      // for each word candidate
-      while(tokens.hasMoreTokens) {
-        // normalize word to lower case
-        val word = tokens.nextToken.toLowerCase
-        if (!stopWords.contains(word) && wordPattern.matcher(word).matches) {
-          // word candiate is not a stop word and matches the word pattern
-          uniqueWords.add(word)
-        }
-      }
-
-      // emit all words that occurred at least once
-      for(w <- uniqueWords) {
-        out.collect( (w, 1) )
-      }
-    }
-  }
+  def main(args: Array[String]) {
 
- }
+    // parse parameters
+    val params = ParameterTool.fromArgs(args)
+    val input = params.getRequired("input")
+
+    // set up the execution environment
+    val env = ExecutionEnvironment.getExecutionEnvironment
 
+    // function returns true if string is not a stop word and matches the word pattern regex
+    val isWord = (s : String) => !STOP_WORDS.contains(s) && WORD_PATTERN.findFirstIn(s).isDefined
+
+    // read messageId and body field of the input data
+    val mails = env.readCsvFile[(String, String)](
+      input,
+      lineDelimiter = MBoxParser.MAIL_RECORD_DELIM,
+      fieldDelimiter = MBoxParser.MAIL_FIELD_DELIM,
+      includedFields = Array(0,4)
+    ) map (m => (m._1, m._2.toLowerCase.split("\\W+")  // convert message to lower case and split on word boundary
+      .filter(s => isWord(s)))) // retain only those strings that are words
+
+    // count mails in data set
+    val cnt = mails.count
+
+    // mails is a DataSet[String, Array[String]]
```

Comment:
Can you remove this comment and add the type to `val mails: DataSet[(String, Array[String)]`?